CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -525,6 +525,50 @@ function Humanoid:getCurrentMood()
   end
 end
 
+--! Collect the objects in a rectangle around the humanoid within the same room.
+--!param radius (int) Number of tiles distance from the humanoid to search.
+--!param object_spec (str or array str) Name(s) of objects to find.
+--!return (array objs, or table str -> array objects) Found requested objects.
+--  One array is returned if an object name was supplied, else a table with
+--  object names to array of objects.
+function Humanoid:findObjectsInSquare(radius, object_spec)
```

Comment:
`findObjectsWithinRadius`?

Reply:
Within Radius implies a circle; Manhattan Distance implies a diamond, this is a square. Maybe the radius parameter name is wrong but I don't know the right term for this.

Reply:
Distance/range for the parameter?

Reply:
I'll switch to `size`. That includes the tile of the humanoid itself, so it needs a little more explanation, and some additional computations to handle even numbers correctly.